domain/application/service/status.go

```diff
@@ -148,9 +171,37 @@ func encodeUnitWorkloadStatus(s *status.StatusInfo) (*application.StatusInfo[app
 	}
 
 	return &application.StatusInfo[application.UnitWorkloadStatusType]{
-		Status:  status,
+		Status:  encodedStatus,
 		Message: s.Message,
 		Data:    bytes,
 		Since:   s.Since,
 	}, nil
 }
+
+// decodeCloudContainerStatus converts a db status info to a core status info.
+// Implicitly validates the status.
+func decodeUnitWorkloadStatus(s *application.StatusInfo[application.UnitWorkloadStatusType]) (*status.StatusInfo, error) {
+	if s == nil {
+		return nil, nil
+	}
+
+	decodedStatus, err := decodeUnitWorkloadStatusType(s.Status)
+	if err != nil {
+		return nil, err
+	}
+
+	data := make(map[string]interface{})
```

Comment:
```suggestion
	var data map[string]interface{}
```

You don't need to make a map, this can just be a nil map.

Reply:
Sure thing